src/java/picard/sam/markduplicates/util/OpticalDuplicateFinder.java

```diff
@@ -96,13 +99,14 @@ public int compare(final PhysicalLocation lhs, final PhysicalLocation rhs) {
             for (int j = i + 1; j < length; ++j) {
                 final PhysicalLocation rhs = list.get(j);
 
-                if (opticalDuplicateFlags[j]) continue;
+                if ((rhs == keeper && opticalDuplicateFlags[i]) || opticalDuplicateFlags[j]) continue;
                 if (lhs.getReadGroup() != rhs.getReadGroup()) continue outer;
                 if (lhs.getTile() != rhs.getTile()) continue outer;
                 if (rhs.getX() > lhs.getX() + this.opticalDuplicatePixelDistance) continue outer;
 
                 if (Math.abs(lhs.getY() - rhs.getY()) <= this.opticalDuplicatePixelDistance) {
-                    opticalDuplicateFlags[j] = true;
+                    if (rhs == keeper) opticalDuplicateFlags[i] = true;
```

Comment:
can you break here since now `(rhs == keeper && opticalDuplicateFlags[i])` will always be true, which causes the continue statement on line `102` to always be invoked?